esp-hal/src/pcnt/mod.rs

```diff
@@ -147,11 +145,25 @@ pub struct PCNT<'d> {
 
 impl<'d> PCNT<'d> {
     /// Return a new PCNT
-    pub fn new(_instance: impl Peripheral<P = crate::peripherals::PCNT> + 'd) -> Self {
+    pub fn new(
+        _instance: impl Peripheral<P = crate::peripherals::PCNT> + 'd,
+        interrupt: Option<crate::interrupt::InterruptHandler>,
```

Comment:
Please import types instead of having full paths all over the place. Just adding the following considerably simplifies this function:

```rust
use crate::{
    interrupt::{self, InterruptHandler},
    peripherals::Interrupt,
};
```